### PR TITLE
feat(PluginManager): Make possbile to pass plugins list, deprecate contexts

### PR DESCRIPTION
**Pull request by yehor-manzhula**

### CODE REVIEW

File: src/pluginManager/browser.ts [hunk 1]
```diff
@@ -19,8 +19,13 @@ import { consoleReportingPlugin, reportingPluginsWrapper } from '../plugins/repo
...
 
 export default class PluginManager extends CommonPluginManager implements PluginManagerPublicApi {
-    constructor(...contexts: Array<Context>) {
-        super(['transitionHooks', 'reporting'], ...contexts);
+    /**
+     * @deprecated use: new PluginManager(...plugins: Plugins[]) instead
+     */
+    constructor(...plugins: Context[]);
+    constructor(...plugins: Plugin[]);
+    constructor(...plugins: Plugin[] | Context[]) {
```
Comment (blackrabbit99) [inline src/pluginManager/browser.ts:27]: Plugin[] | Context[] Looks like it duplicates 
```
constructor(...plugins: Context[]);
    constructor(...plugins: Plugin[]);
```

File: src/pluginManager/browser.ts [hunk 2]
```diff
@@ -19,8 +19,13 @@ import { consoleReportingPlugin, reportingPluginsWrapper } from '../plugins/repo
...
+        super(['transitionHooks', 'reporting'], ...plugins);
```
Comment (blackrabbit99) [inline src/pluginManager/browser.ts:28]: `['transitionHooks', 'reporting']` should we move to config ?

File: src/pluginManager/browser.ts (hunk 1 above)
Comment (yehor-manzhula) [reply src/pluginManager/browser.ts:27]: No it is not. Required by TS: as far as method overload doesn't allow to have miltiple implementations all possible values should be included for single implementation. See [playground](https://www.typescriptlang.org/play?#code/FAYwNghgzlAEBiB7RAjCAnWBvWxb9gHoAqYvA2Y2AAQBMBTAB3XpAgBd7bYBXKe2ADt6AdwTI06ABQQAXEJ4BbFPXQBKWAEtBUThFrkCxQodghEO9uh4h2iaXNi702gOZqA3KfOXrt+zLygkoq6l4UPs42dg7yzm6wAD4KyqoaWMCmAL7AOaCQMLAAQhhIiNimRKSVVHRMLGyc3HwCwmJlkoEpoRrauvT6NSYRFlH+sU5Wbp7eo1bRAY7BqeoVFBTZuZmgc7AAZsglmAC8QqLiqBhSAOQArNczQA).

File: src/pluginManager/browser.ts (hunk 2 above)
Comment (yehor-manzhula) [reply src/pluginManager/browser.ts:28]: Good idea, but I would like to have separate tiket for that.

**Review by b1ff — APPROVED** [review]

**Pull request merged**